src/engraving/dom/textbase.cpp

```diff
@@ -3315,17 +3315,23 @@ void TextBase::drawEditMode(Painter* p, EditData& ed, double currentViewScaling)
     pen.setJoinStyle(PenJoinStyle::MiterJoin);
     p->setPen(pen);
 
-    // Don't draw cursor if there is a selection
-    if (!cursor->hasSelection()) {
+    // Don't draw cursor if it's dynamic or if there is a selection
+    if (!ed.element->isDynamic() && !cursor->hasSelection()) {
         p->drawRect(cursor->cursorRect());
     }
 
     p->translate(-pos);
     p->setPen(Pen(configuration()->formattingColor(), 2.0 / currentViewScaling)); // 2 pixel pen size
     p->setBrush(BrushStyle::NoBrush);
 
-    double m = spatium();
-    RectF r = canvasBoundingRect().adjusted(-m, -m, m, m);
+    RectF r;
+
+    if (ed.element->isDynamic()) {
+        r = toDynamic(ed.element)->adjustedBoundingRect();
+    } else {
+        double m = spatium();
+        r = canvasBoundingRect().adjusted(-m, -m, m, m);
+    }
```

Comment:
Here applies what I said about checking whether we really need the `adjustedBoundingRect` special case